src/common/SSLUtil.cc

```diff
@@ -29,61 +28,59 @@ namespace ssl_util
 
     void base64_decode(const std::string& in, std::string& out)
     {
-        BIO * bio_64  = BIO_new(BIO_f_base64());
+        const int max_size = 3 * in.length()/4 + 1;
+        auto output        = new unsigned char[max_size];
 
-        BIO * bio_mem_in  = BIO_new(BIO_s_mem());
-        BIO * bio_mem_out = BIO_new(BIO_s_mem());
+        int size = EVP_DecodeBlock(output, reinterpret_cast<const unsigned char *>(in.c_str()), in.length());
 
-        bio_64 = BIO_push(bio_64, bio_mem_in);
-
-        BIO_set_flags(bio_64, BIO_FLAGS_BASE64_NO_NL);
-
-        BIO_write(bio_mem_in, in.c_str(), in.length());
-
-        char inbuf[512];
-        int  inlen;
-
-        while ((inlen = BIO_read(bio_64, inbuf, 512)) > 0)
+        if (size <= 0)
         {
-            BIO_write(bio_mem_out, inbuf, inlen);
+            out.clear();
+            delete[] output;
+
+            return;
         }
 
-        char * decoded_c;
+        /* Subtract padding bytes from |size|. Any more than 2 is malformed. */
+        size_t inlen = in.length();
+        int i = 0;
+        while (in[--inlen] == '=')
+        {
+            --size;
+            if (++i > 2)
+            {
+                out.clear();
+                delete[] output;
 
-        long int size = BIO_get_mem_data(bio_mem_out, &decoded_c);
+                return;
+            }
+        }
 
-        out.assign(decoded_c, size);
+        out.assign(reinterpret_cast<char*>(output), size);
 
-        BIO_free_all(bio_64);
-        BIO_free_all(bio_mem_out);
+        delete[] output;
     }
 
     /* -------------------------------------------------------------------------- */
     /* -------------------------------------------------------------------------- */
 
     int base64_encode(const std::string& in, std::string &out)
     {
-        BIO * bio_64  = BIO_new(BIO_f_base64());
-        BIO * bio_mem = BIO_new(BIO_s_mem());
-
-        BIO_push(bio_64, bio_mem);
+        const int max_size = 4*((in.length()+2)/3) + 1;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

We observed that an integer overflow arising for your program. It happens when the outcome of an arithmetic operation surpasses the maximum representable value for the utilized data type. It can lead to unpredictable behavior, security vulnerabilities, and system crashes. To address this vulnerability, it is essential to perform thorough range checking and validation of input values. Employ data types with adequate size to accommodate the anticipated range of values and implement checks to identify and manage potential overflow conditions.